src/cache/cache.jl

```diff
@@ -70,6 +70,9 @@ struct AtmosCache{
     conservation_check::CONSCHECK
 end
 
+# Allow cache to be moved on the CPU. Used by ClimaCoupler to save checkpoints
+ClimaCore.Adapt.@adapt_structure AtmosCache
```

Comment:
I kind of think that Atmos should just directly depend on Adapt, it's a lightweight package, and code loading like this is a bit confusing (and could depend on ClimaCore versions).

@dennisYatunin, @imreddyTeja, @nefrathenrici, what do you think?

Reply:
I am fine with that

Reply:
Sounds good to me, Adapt does not introduce many dependencies